widgets/src/main/java/org/pentaho/gwt/widgets/client/utils/MenuBarUtils.java

```diff
@@ -0,0 +1,51 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2023 Hitachi Vantara. All rights reserved.
+ */
+
+package org.pentaho.gwt.widgets.client.utils;
+
+import com.google.gwt.user.client.ui.DecoratedPopupPanel;
+import com.google.gwt.user.client.ui.MenuBar;
+import com.google.gwt.user.client.ui.MenuItem;
+import com.google.gwt.user.client.ui.UIObject;
+
+import java.util.List;
+
+public class MenuBarUtils {
+  // "Static" class. Prevent instantiation.
+  private MenuBarUtils() {
+  }
+
+  // Access to private field allItems
+  public static native List<UIObject> getAllItems( MenuBar menuBar ) /*-{
+    return menuBar.@com.google.gwt.user.client.ui.MenuBar::allItems;
+  }-*/;
+
+  // Access to private field items
+  public static native List<MenuItem> getItems( MenuBar menuBar ) /*-{
```

Comment:
Why are we using JSNI for purely java code?

Can't we just use the java function directly https://www.gwtproject.org/javadoc/latest/com/google/gwt/user/client/ui/MenuBar.html#getItems--?

```
public static native List<MenuItem> getItems( MenuBar menuBar ) {
   return menuBar.getItems()
}
```

Reply:
This is a GWT hack for accessing private members. Akin to using reflection for same purposes in Java land.
See https://groups.google.com/g/google-web-toolkit/c/gQYJ7j3O1FA.

Reply:
cool